### PR TITLE
ci: docker automation

### PR DESCRIPTION
**Pull request by gszr**
### Summary

- Add script to automate Docker images build and release
- Integrate Docker automation into Circle CI

### CODE REVIEW

File: tools/releases/docker.sh [hunk 1]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+
+set -e
+
+KUMA_DOCKER_REPO="kong-docker-kuma-tests-docker.bintray.io"
```
Comment (dhruv) [inline tools/releases/docker.sh]: Do we need to change this before merging?

File: tools/releases/docker.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+
+function msg_err() {
+  msg_red $@
+  exit 1
+}
+
+
+function build {
+  msg "Building kuma-cp..."
```
Comment (dhruv) [inline tools/releases/docker.sh]: What do you think about parameterizing these 4 blocks?

File: tools/releases/docker.sh [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+function docker_logout {
+  docker logout $KUMA_DOCKER_REPO
+}
+
+
+function push {
+  docker_login
+
+  msg "Pushing kuma-cp..."
```
Comment (dhruv) [inline tools/releases/docker.sh]: ditto. what do you think about parameterizing these 4 blocks?

File: tools/releases/dockerfiles/Dockerfile.kuma-dp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+FROM envoyproxy/envoy-alpine:latest
```
Comment (dhruv) [inline tools/releases/dockerfiles/Dockerfile.kuma-dp]: Please leave a comment here about why we are using this as the base

File: tools/releases/docker.sh (hunk 1 above)
Comment (gszr) [reply tools/releases/docker.sh]: We do

File: tools/releases/dockerfiles/Dockerfile.kuma-cp [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FROM alpine:3.10
```
Comment (yskopets) [inline tools/releases/dockerfiles/Dockerfile.kuma-cp]: Docker image should run under a non-`root` account, e.g. `nobody` (`65534:65534`) on `alpine`

File: tools/releases/dockerfiles/Dockerfile.kuma-injector [hunk 6]
```diff
@@ -0,0 +1,9 @@
+FROM alpine:3.10
```
Comment (yskopets) [inline tools/releases/dockerfiles/Dockerfile.kuma-injector]: Docker image should run under a non-`root` account, e.g. `nobody` (`65534:65534`) on `alpine`

File: tools/releases/dockerfiles/Dockerfile.kuma-tcp-echo [hunk 7]
```diff
@@ -0,0 +1,9 @@
+FROM alpine:3.10
```
Comment (yskopets) [inline tools/releases/dockerfiles/Dockerfile.kuma-tcp-echo]: Docker image should run under a non-`root` account, e.g. `nobody` (`65534:65534`) on `alpine`

File: tools/releases/dockerfiles/Dockerfile.kuma-dp [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# using Envoy's base to inherit the Envoy binary
```
Comment (yskopets) [inline tools/releases/dockerfiles/Dockerfile.kuma-dp]: Docker image should run under a non-`root` account, e.g. `nobody` (`65534:65534`) on `alpine`

File: tools/releases/docker.sh [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+function docker_logout {
+  docker logout $KUMA_DOCKER_REPO
+}
+
+
+function push {
+  docker_login
+
+  for component in "${KUMA_COMPONENTS[@]}"; do
+    msg "Pushing kuma-cp..."
```
Comment (yskopets) [inline tools/releases/docker.sh]: Can we change the message to `Pushing $component ...` ?

**Pull request merged**